### PR TITLE
FUSETOOLS2-1091 - adding a progress popup for the refresh classpath c…

### PR DESCRIPTION
**Pull request by lhein**
…ommand

Signed-off-by: Lars Heinemann <lhein.smx@gmail.com>

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -175,7 +175,16 @@ export async function activate(context: vscode.ExtensionContext) {
...
 		vscode.commands.registerCommand('camelk.integrations.selectFirstNode', () => { selectFirstItemInTree();});
-		vscode.commands.registerCommand('camelk.classpath.refresh', async (uri:vscode.Uri) => { await downloadSpecificCamelKJavaDependencies(context, uri, mainOutputChannel)});
+		vscode.commands.registerCommand('camelk.classpath.refresh', async (uri:vscode.Uri) => {
+			vscode.window.withProgress({
+				location: vscode.ProgressLocation.Notification,
+				title: "Updating the classpath...",
+				cancellable: false
+			}, async (progress) => {
+				progress.report({ increment: -1 });
+				return downloadSpecificCamelKJavaDependencies(context, uri, mainOutputChannel);
```
Comment (apupier) [inline src/extension.ts:185]: It is reporting only for a subpart of the refresh.
There are 2 parts for the user to really benefits from the classpath refresh:
- the update of the folder containing all dependencies that will be part of the classpath
- the Java Language Server reading the folder and updating the classpath and recompiling.

We are not waiting here that the Java Language Server is picking the new files

File: src/extension.ts [hunk 2]
```diff
@@ -175,7 +175,16 @@ export async function activate(context: vscode.ExtensionContext) {
 
 		vscode.commands.registerCommand('camelk.integrations.createNewIntegrationFile', async (...args:any[]) => { await NewIntegrationFileCommand.create(args);});
 		vscode.commands.registerCommand('camelk.integrations.selectFirstNode', () => { selectFirstItemInTree();});
-		vscode.commands.registerCommand('camelk.classpath.refresh', async (uri:vscode.Uri) => { await downloadSpecificCamelKJavaDependencies(context, uri, mainOutputChannel)});
+		vscode.commands.registerCommand('camelk.classpath.refresh', async (uri:vscode.Uri) => {
+			vscode.window.withProgress({
+				location: vscode.ProgressLocation.Notification,
+				title: "Updating the classpath...",
```
Comment (apupier) [inline src/extension.ts:181]: would be nice to provide a more precise message something like:
"Updating the classpath of the Camel K Integration file..."

or maybe as a first iteration, as only the download of the libraries is awaited, something like: "Setting up dependencies for the Camel K integration file..."

File: src/extension.ts (hunk 1 above)
Comment (lhein) [reply src/extension.ts:185]: That seems to be an issue of the underlying functionality then. I am waiting until the command promise is resolved. If that is not enough, then there seems to be a logical issue in how we call / trigger the java lsp cp refresh as it seems we are not waiting for it to finish before resolving the command.

Comment (lhein) [reply src/extension.ts:185]: Ok, we are basically lacking a handle on the Java LSP classpath update and hence can't wait for it to be done in a nice way. I am going to rename the text of the progress popup so it becomes more obvious we are only covering the Camel K dependencies update. This needs to be revisited once Java LSP provides an API to call the refresh.

**Review by apupier — CHANGES_REQUESTED** [review]
readme and changelog to be updated as they were mentioning that no progress report was provided

**Comment by apupier** [comment]
@lhein please rebase on master. it should avoid most of the flaky tests, and for the remaining ones, there will be more details at least.

**Comment by apupier** [comment]
current error:

```
rejected promise not handled within 1 second: Error [ERR_STREAM_WRITE_AFTER_END]: write after end

stack trace: Error [ERR_STREAM_WRITE_AFTER_END]: write after end
	at writeAfterEnd (_stream_writable.js:266:14)
	at Socket.Writable.write (_stream_writable.js:315:5)
	at /home/circleci/.vscode/extensions/redhat.java-0.79.0/dist/extension.js:2:882887
	at new Promise (<anonymous>)
	at c.write (/home/circleci/.vscode/extensions/redhat.java-0.79.0/dist/extension.js:2:882805)
	at _.doWrite (/home/circleci/.vscode/extensions/redhat.java-0.79.0/dist/extension.js:2:872391)
	at /home/circleci/.vscode/extensions/redhat.java-0.79.0/dist/extension.js:2:872286

      1) Completes additional dependencies
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -10,8 +10,8 @@ All notable changes to the "vscode-camelk" extension will be documented in this
   - If mistakenly called with Camel K 1.3.2-, need to restart VS Code for basic dependencies to be available again.
   - It supports modeline dependencies notation from local build. See [apache/camel-k#2213](https://github.com/apache/camel-k/issues/2213)
   - A single classpath is provided. It means that refresh command needs to be called when switching between Integration file written in Java that does not have the same dependencies.
-  - There is no progress indicator. Please be patient. The first time may take several minutes on a slow network.
-- Provide telemetry information when using `Start Apache Camel K integration` command about language of the file deployed and the kind of deployment used (basic, dev, ...)
+  - There is a progress indicator for updating the Camel K file dependencies but not for the underlying Java classpath update. Please be patient. The first time may take several minutes on a slow network.
+  - Provide telemetry information when using `Start Apache Camel K integration` command about language of the file deployed and the kind of deployment used (basic, dev, ...)
```
Comment (apupier) [inline CHANGELOG.md:14]: changelog to be updated as 0.0.24 was released

**Pull request closed without merging**